src/compile/mark/mark.ts

```diff
@@ -239,8 +239,8 @@ function getStackByFields(model: UnitModel) {
         const fieldDef: FieldDef = channelEncoding;
         const scale = model.scale(channel);
         fields.push(field(fieldDef, {
-          binSuffix: scale && scale.type === ScaleType.ORDINAL ? '_range' : '_start'
-        }));
+          binSuffix: scale && scale.type === ScaleType.ORDINAL ? 'range' : 'start'
```

Comment:
Also -- I don't understand why this check of type is here, because a **very** similar check is done in field() (within fielddef.ts). 

I previous attempted to get rid of it here and get it to be triggered there, but it caused tests to fail. Rather than investigate, I just left it. Factoring this out into FieldDef is probably worth considering.

Furthermore, I wasn't able to find a single example of when _both_ binSuffix and suffix were being used for the same fieldDef, which leads me to believe that we should just have one? I tried implementing this too, and it broke tests because if you pass in suffix it gets set immediately in field(), while binSuffix is only set when `!opt.nofn && fieldDef.bin`. Rather than add a new flag, I left in binSuffix. Wondering why this was done this way initially though.